src/main.rs

```diff
@@ -300,10 +309,118 @@ fn main2() -> Result<(), Box<dyn Error>> {
         log::info!("Elapsed: {:.2?} rows: {}", elapsed, example_num);
 
         if let Some(filename) = final_regressor_filename {
-            save_sharable_regressor_to_filename(filename, &mi, &vw, sharable_regressor, quantize_weights)
-                .unwrap()
+            save_sharable_regressor_to_filename(
+                filename,
+                &mi,
+                &vw,
+                sharable_regressor,
+                quantize_weights,
+            )
+            .unwrap()
         }
     }
 
     Ok(())
 }
+
+#[cfg(test)]
```

Comment:
Not sure I like these very specific tests in the main module... Can we extract the compression logic into another module?

Reply:
Separated to `buffer_handler.rs` + applied `rustfmt` to align the repo